v4/core/utils.go

```diff
@@ -179,3 +179,57 @@ func PrettyPrint(result interface{}, resultName string) {
 func GetCurrentTime() int64 {
 	return time.Now().Unix()
 }
+
+// ConvertSlice Marshals 'slice' to a json string, performs
+// string manipulation on the resulting string, and converts
+// the string to a '[]string'. If 'slice' is nil, not a 'slice' type,
+// or an error occurred during conversion, an empty slice will be returned.
+func ConvertSlice(slice interface{}) (s []string) {
+	s = []string{}
+	inputIsSlice := false
+
+	if IsNil(slice) {
+		return
+	}
+
+	// Reflect on 'slice' to validate the input is in fact a slice
+	rResultType := reflect.TypeOf(slice)
+
+	switch rResultType.Kind() {
+	case reflect.Slice:
+		inputIsSlice = true
+	default:
+	}
+
+	// If it's not a slice or if it's an empty slice, just return
+	if !inputIsSlice || reflect.ValueOf(slice).Len() == 0 {
+		return
+	}
+
+	jsonBuffer, err := json.Marshal(slice)
+	if err != nil {
+		return
+	}
+
+	// Convert the json buffer to a string and remove any escaped quotes
+	jsonString := strings.Replace(string(jsonBuffer), "\"", "", -1)
```

Comment:
I think after serializing to a JSON string, we want to retain any escaped quotes, and only remove any quotes surrounding the value of each slice element.   IOW, we need to be really careful about what we're removing so as not to remove any characters that were actually contained in the elements of the input slice.

Reply:
This took me a while to figure out, but I added an alternative solution that only removes the _starting_ and _ending_ surrounding quotes added by the json marshaler, if present.